docs/juliacon_2023/juliacon_2023_presentation.md

```diff
@@ -0,0 +1,440 @@
+---
+marp: true
+---
+
+# HDF5.jl: Hierarchical Data Storage for Julia
+
+Mark Kittisopikul (HHMI), Simon Byrne (Caltech), Mustafa Mohamad (UCalgary)
+
+---
+
+# What is HDF5?
+
+HDF5 stands for Hierarchial Data Format version 5 and is maintained by The HDF Group, formerly part of the National Center for Supercomputing Appplications (NCSA).
+
+* HDF5 is a file format with an open specification.
+* HDF5 is a C Library and API.
+* HDF5 is a data model.
+
+---
+
+# When to use HDF5
+
+* Store numeric array and attributes in nested groups.
+* Use it when you want to compactly store binary data.
+
+## When not to use HDF5
+* You have arrays of variable-length strings. Used fixed lengths strings instead.
+* You have tables of heterogeneous data. Consider using columnar layouts. Other formats are more optimized for tables.
+
+---
+
+# Related formats
+
+HDF5 is used as a base for other formats
+* NetCDF - Network Common Data Form v4 (Unidata, UCAR)
+* MAT - MATLAB data files v7.3+
+* PyTables - Pandas
+* JLD/JLD2 - Julia Data Format
+
+---
+
+# HDF5 Specification
+
+The HDF5 specification is open and freely available.
+
+![width:600px](images/specification_toc.png)
+
+https://docs.hdfgroup.org/hdf5/v1_14/_f_m_t3.html
+
+---
+
+# What is HDF5.jl?
+
+HDF5.jl is a wrapper around the HDF5 C Library.
+
+It consists of
+* A low level interface, a direct mapping to the C API
+* A mid level interface, lightweight helpers
+* A high level interface, a Julia API
+<!-- give examples ? -->
+
+---
+
+# Related Julia Packages
+
+* HDF5_jll.jl, C Library from HDF Group (dependency of HDF5.jl)
+* MAT.jl, MATLAB files (depends on HDF5.jl)
+* JLD.jl, Julia Data Format (depends on HDF5.jl)
+* JLD2.jl, Julia Data Format 2: pure Julia implementation of a subset of HDF5
+* NetCDF.jl & NCDatasets.jl: wrappers for the NetCDF C library, which incorporates HDF5
+
+---
+
+# HDF5.jl Early and Recent Contributors
+
+* There are many contributors
+* Konrad Hisen initiated Julia's support for HDF5
+* Tim Holy and Simon Kornblith were the initial primary authors
+* Tom Short, Blake Johnson, Isaih Norton, Elliot Saba, Steven Johnson, Mike Nolta, Jameson Nash
+* Justin Willmert improved many aspects C to Julia API interface
+* Other recent contributors: t-bltg, Henrik Ranocha, Nathan Zimmerberg, Joshua Lampert, Tamas Gal, David MacMahon, Juan Ignacio Polanco, Michale Schlottke-Lakemper, linwaytin, Dmitri Iouchtchenko, Lorenzo Van Munoz, Jared Wahlstrand, Julian Samaroo, machakann, James Hester, Ralph Kube, Kristoffer Carlsson
```

Comment:
Two small typos: It's **Hendrik** Ranocha and **Michael** Schlottke-Lakemper :smile: